File: linked_lists/merge_in_between_linked_lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeInBetween(ListNode* list1, int a, int b, ListNode* list2) {
        cin.tie(nullptr);
	    cout.tie(nullptr);
	    ios_base::sync_with_stdio(false);

        ListNode* curr = list1;

        for (int i = 0; i < a - 1; ++i) curr = curr->next;
        ListNode* tail1 = curr;

        for (int i = a - 1; i < b + 1; ++i) curr = curr->next;
        ListNode* tail2 = curr;

        tail1->next = list2;
        curr = list2->next;
        while (curr->next) curr = curr->next;
        curr->next = tail2;

        return list1;
    }
};

File: linked_lists/merge_in_between_linked_lists.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode MergeInBetween(ListNode list1, int a, int b, ListNode list2) {
        ListNode current = list1;

        for (int i = 0; i < a - 1; ++i) current = current.next;
        ListNode tail1 = current;

        for (int i = a - 1; i < b + 1; ++i) current = current.next;
        ListNode tail2 = current;

        tail1.next = list2;
        current = list2;
        while (current.next != null) current = current.next;
        current.next = tail2;

        return list1;
    }
}
